metadata-integration/java/datahub-schematron/lib/src/test/java/io/datahubproject/schematron/converters/avro/AvroSchemaConverterTest.java

```diff
@@ -0,0 +1,942 @@
+        false,
+        false,
+        new SchemaFieldDataType().setType(SchemaFieldDataType.Type.create(new NumberType())));
+    assertSchemaField(
+        schema.getFields().get(7),
+        "[version=2.0].[type=PrimitiveType].[type=double].doubleField",
+        "double",
+        false,
+        false,
+        new SchemaFieldDataType().setType(SchemaFieldDataType.Type.create(new NumberType())));
+    assertSchemaField(
+        schema.getFields().get(8),
+        "[version=2.0].[type=PrimitiveType].[type=string].stringField",
+        "string",
+        false,
+        false,
+        new SchemaFieldDataType().setType(SchemaFieldDataType.Type.create(new StringType())));
+    assertSchemaField(
+        schema.getFields().get(9),
+        "[version=2.0].[type=PrimitiveType].[type=boolean].booleanField",
+        "boolean",
+        false,
+        false,
+        new SchemaFieldDataType().setType(SchemaFieldDataType.Type.create(new BooleanType())));
+    assertSchemaField(
+        schema.getFields().get(10),
+        "[version=2.0].[type=PrimitiveType].[type=int].nullableIntField",
+        "int",
+        true,
+        false,
+        new SchemaFieldDataType().setType(SchemaFieldDataType.Type.create(new NumberType())));
+    assertSchemaField(
+        schema.getFields().get(11),
+        "[version=2.0].[type=PrimitiveType].[type=long].nullableLongField",
+        "long",
+        true,
+        false,
+        new SchemaFieldDataType().setType(SchemaFieldDataType.Type.create(new NumberType())));
+    assertSchemaField(
+        schema.getFields().get(12),
+        "[version=2.0].[type=PrimitiveType].[type=string].nullableStringField",
+        "string",
+        true,
+        false,
+        new SchemaFieldDataType().setType(SchemaFieldDataType.Type.create(new StringType())));
+    assertSchemaField(
+        schema.getFields().get(13),
+        "[version=2.0].[type=PrimitiveType].[type=enum].status",
+        "Enum",
+        false,
+        false,
+        new SchemaFieldDataType().setType(SchemaFieldDataType.Type.create(new EnumType())));
+  }
+
+  @Test(groups = "basic")
+  void testComplexMaps() throws IOException {
+    SchemaMetadata schema =
+        avroSchemaConverter.toDataHubSchema(
+            readAvroSchema("complex_maps.avsc"), false, false, dataPlatformUrn, null);
+
+    schema.getFields().forEach(System.out::println);
+
+    assertEquals(schema.getFields().size(), 15);
+
+    assertSchemaField(
+        schema.getFields().get(0),
+        "[version=2.0].[type=MapType].[type=map].mapOfString",
+        "map<string,string>",
+        false,
+        false,
+        new SchemaFieldDataType()
+            .setType(
+                SchemaFieldDataType.Type.create(
+                    new MapType().setKeyType("string").setValueType("string"))));
+    assertSchemaField(
+        schema.getFields().get(1),
+        "[version=2.0].[type=MapType].[type=map].[type=ComplexType].mapOfComplexType",
+        "ComplexType",
```

Comment:
should native type be `map<string,ComplexType>`?

Reply:
I think this makes sense